added bounty hunter program doc (#4847)
treeverse/lakeFS: docs/lakeFSBountyHunterProgram.md

> lakeFS Bounty Hunter Program

Reviewer: this is missing a header that connects all our new pages together. 
@hiccups28 take a look at this one as an example - https://raw.githubusercontent.com/treeverse/lakeFS/master/docs/contributing.md

> lakeFS Bounty Hunter Program

Reviewer 2: These programs called "Bug Bounty" without the hunter

> lakeFS Bounty Hunter Program is a community program to encourage lakeFS Active OSS users to engage with the source code and improve lakeFS capabilities. The program is designed to reward community members who actively find bugs and suggest product improvements within lakeFS. Through this program, we hope to empower and recognize veteran lakeFS users. 

Reviewer 3: No need to capitalize "Active" throughout this doc.

*Video going through the bounty hunter program*

> Getting Started

Reviewer 3: Please use markdown headings.

> Before we get started, please see the requirements to join the program:

Reviewer 3: ```suggestion
### Requirements
```

> New Bounty Users must have an Active lakeFS OSS installation 
> New Users must be an active OSS users for at least 1 month
> Must be a member of lakeFS Community Slack

Reviewer 3: ```suggestion
- To join the program, the user must have a lakeFS installation that has been active for at least 1 month.
- The user must be a member of the lakeFS community Slack.
```

> To be an active OSS users, must be apart of 6 versions,

Reviewer 3: What?

To become a new lakeFS Bounty Hunter, please apply here.

First & last name
Github Username
Email
Company Name & Position
Do you use lakeFS for work?
lakeFS Active OSS installation 

> New Users will receive an email to verify that they have been accepted to the Bounty Hunter Program and will be given a link to access the bounties.

Reviewer 3: ```suggestion
New Users will receive an email to confirm that they have been accepted to the Bounty Hunter Program and will be given a link to access the bounties.
```

lakeFS Bounty Hunter Dashboard

New Bounty Users will be given a link in their email to access the Bounty Project under lakeFS Git repository. 

The Dashboard will list out the following:
New Issues
Issues in progress
Completed Issues

New Issues will be assigned to Bounty Users as a first come first serve basis. Bounty Users can also open new issues if they find any along the way. Issue cards will always contain the following information:

lakeFS Version
Type of Installation 
Description
Error Code
How to reproduce the error

Bounty Review Process

> All new issues submitted to the lakeFS bounty program will be reviewed by a lakeFS Developer. Here are the list of submission acceptable for review:

Reviewer 3: ```suggestion
All new issues submitted to the lakeFS bounty program will be reviewed by a lakeFS developer. Here is the list of submission types acceptable for review:
```
